x-pack/plugins/maps/public/shared/layers/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -87,7 +87,8 @@ export class EMSTMSSource extends AbstractTMSSource {
   }
 
   async getDisplayName() {
-    return this._descriptor.id;
+    const emsTmsMeta = await this._getEmsTmsMeta();
```

Comment:
`_getEmsTmsMeta` can throw an exception when meta can not be found. Wrap in try/catch and return id if name can not be loaded. Similiar to how name is fetched for elasticsearch sources https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/shared/layers/sources/es_source.js#L268